### DISCUSSION TITLE
onlineManager on React Native

### DISCUSSION THREAD

**Discussion by nihilenz** [TanStack/query#3747] [Q&A]
Hi, I followed the page https://react-query.tanstack.com/react-native, installed `@react-native-community/netinfo` (`^9.1.0`) and added the recommended code into `App.tsx`:
```
import NetInfo from '@react-native-community/netinfo'
import { onlineManager } from 'react-query'

onlineManager.setEventListener(setOnline => {
  return NetInfo.addEventListener(state => {
    setOnline(state.isConnected)
  })
})
```

When I check the state with `onlineManager.isOnline()` I always got `true`.

What am I doing wrong?
Thanks!

**Comment by joaovitorzv** [comment]
you might want to use the "truthy" operator `!!` when setting your state 
```javascript
setOnline(!!state.isConnected);
```
